specs/signing-and-verification-workflow.md

```diff
@@ -8,19 +8,19 @@ The user wants to sign an OCI artifact and push the signature to a repository.
 
 ### Signing Prerequisites
 
-- User has access to the signing certificate and private key or a remote signing service through a notation plug-in.
+- User has access to the signing certificate and private key or a remote signing service through a notation plugin.
 
 ### Signing Steps
 
 1. **Generate signature:** Using notation CLI or any other compliant signing tool, sign the OCI artifact. The signing tool should follow the following guideline.
     1. Verify that the signing certificate is valid and satisfies [certificate requirements](./signature-specification.md#certificate-requirements).
     1. Verify that the signing algorithm satisfies [algorithm requirements](./signature-specification.md#signature-algorithm-requirements).
     1. Generate signature.
-        1. Generate signature using signature formats specified in [supported signature envelopes](./signature-specification.md#supported-signature-envelopes). Also, as part of this step, the user-defined/supplied custom attributes should be added to the annotations of the signature's descriptor.
-        1. If the user wants to timestamp the signature, obtain an [RFC-3161](https://datatracker.ietf.org/doc/html/rfc3161.html) compliant timestamp for the signature generated in the previous step. Otherwise, continue to the next step.
+        1. Generate signature envelope using signature formats specified in [supported signature envelopes](./signature-specification.md#supported-signature-envelopes). Also, as part of this step, the user-defined/supplied custom attributes should be added to the annotations of the signature's descriptor.
+        1. If signing scheme is [`notary.x509`](./signing-scheme.md/#notaryx509) and timestamp countersignature is demanded, extract the [primitive signature](./signature-specification.md#signature-envelope) (the digital signature computed on payload and signed attributes) from the signature envelope generated in the previous step. Compute hash of the signature (the hash algorithm SHOULD be deduced from signing certificate's public key) and send it to a [RFC-3161](https://datatracker.ietf.org/doc/html/rfc3161.html) compliant `Timestamp Authority (TSA)` for timestamping. Otherwise, continue to the next step. 
```

Comment:
> `Timestamp Authority (TSA)`

TSA is not a code term. You can use `*` instead. For example, *Timestamp Authority (TSA)*.